src/Microsoft.Azure.WebJobs.Host/Indexers/FunctionIndexingException.cs

```diff
@@ -3,17 +3,68 @@
 
 using System;
 using System.Diagnostics.CodeAnalysis;
+using System.Runtime.Serialization;
 
 namespace Microsoft.Azure.WebJobs.Host.Indexers
 {
-    [SuppressMessage("Microsoft.Design", "CA1064:ExceptionsShouldBePublic")]
+    /// <summary>
+    /// Exception that occurs when the <see cref="JobHost"/> encounters errors when trying
+    /// to index job methods on startup.
+    /// </summary>
     [SuppressMessage("Microsoft.Design", "CA1032:ImplementStandardExceptionConstructors")]
     [Serializable]
-    internal class FunctionIndexingException : Exception
+    public class FunctionIndexingException : Exception
     {
+        /// <summary>
+        /// Constructs a new instance.
+        /// </summary>
+        /// <param name="methodName">The name of the method in error.</param>
+        /// <param name="innerException">The inner exception.</param>
         public FunctionIndexingException(string methodName, Exception innerException)
             : base("Error indexing method '" + methodName + "'", innerException)
         {
+            MethodName = methodName;
+        }
+
+        /// <summary>
+        /// Constructs a new instance.
+        /// </summary>
+        /// <param name="info">The <see cref="SerializationInfo"/>.</param>
+        /// <param name="context">The <see cref="StreamingContext"/>.</param>
+        protected FunctionIndexingException(SerializationInfo info, StreamingContext context) : base(info, context)
+        {
+            if (info == null)
+            {
+                throw new ArgumentNullException("info");
+            }
```

Comment:
How easily can we just yank that rule that requires these useless checks? I'm guessing the amount of time we spend adding those checks for all new code adds up to much more than it would take to fix it. And I really don't like bloating the IL.

I'll keep bringing this one up! :)